OpenRA.Mods.Common/Widgets/Logic/MapGeneratorLogic.cs

```diff
@@ -390,48 +401,61 @@ void OnClick()
 			}
 		}
 
+		System.Threading.CancellationTokenSource cancelationTokenSource;
+
 		void GenerateMap()
 		{
-			generating = true;
-			onGenerate(null, null);
+			var currentGeneration = ++generationCounter;
+
 			failed = false;
+			onGenerate(null, null);
 			preview.Clear();
+
+			cancelationTokenSource?.Cancel();
+			cancelationTokenSource = new System.Threading.CancellationTokenSource();
+			var cancelationToken = cancelationTokenSource;
```

Comment:
Spelling nitpick: `cancellation`

Was `var cancelationToken = cancelationTokenSource;` supposed to be `var cancellationToken = cancellationTokenSource.Token;`?